python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -786,7 +789,9 @@ def _prepare_for_direct_get(
             parameters=parameters,
         )
 
-    def _prepare_for_put(self, inMemoryDataset: Any, ref: DatasetRef) -> tuple[Location, Formatter]:
+    def _prepare_for_put(
+        self, inMemoryDataset: Any, ref: DatasetRef
+    ) -> tuple[Location, Formatter | FormatterV2]:
```

Comment:
Could you do the wrapping of `Formatter` into `FormatterV1inV2` earlier (i.e. in this mehtod), to simplify downstream code?